internal/snmp/translate.go

```diff
@@ -150,20 +150,24 @@ func SnmpTranslateCall(oid string) (mibName string, oidNum string, oidText strin
 		// for example RFC1213-MIB::sysUpTime.0
 		s := strings.Split(oid, "::")
 		// node becomes sysUpTime.0
-		node := s[1]
-		if strings.ContainsAny(node, ".") {
-			s = strings.Split(node, ".")
-			// node becomes sysUpTime
-			node = s[0]
-			end = "." + s[1]
-		}
+		if s[1] != "" {
+			node := s[1]
+			if strings.ContainsAny(node, ".") {
+				s = strings.Split(node, ".")
+				// node becomes sysUpTime
+				node = s[0]
+				end = "." + s[1]
+			}
 
-		out, err = gosmi.GetNode(node)
-		if err != nil {
-			return oid, oid, oid, oid, err
-		}
+			out, err = gosmi.GetNode(node)
+			if err != nil {
+				return oid, oid, oid, oid, err
+			}
 
-		oidNum = "." + out.RenderNumeric() + end
+			oidNum = "." + out.RenderNumeric() + end
+		} else {
+			return "", oid, oid, oid, fmt.Errorf("can not parse %v\n", oid)
+		}
```

Comment:
Please invert the logic to save some indention...
```golang
		s := strings.Split(oid, "::")
		// node becomes sysUpTime.0
		if s[1] == "" {
			return "", oid, oid, oid, fmt.Errorf("cannot parse %v\n", oid)
		}
		node := s[1]
		if strings.ContainsAny(node, ".") {
			s = strings.Split(node, ".")
			// node becomes sysUpTime
			node = s[0]
			end = "." + s[1]
		}

		out, err = gosmi.GetNode(node)
		if err != nil {
			return oid, oid, oid, oid, err
		}

		oidNum = "." + out.RenderNumeric() + end
```

Another thing is that you split unconditionally, that is, if there are multiple (`::` or `.`) separators, you will only process the first and the second element and ignore all others. Maybe it would be better to explicitly use [`strings.SplitN(..., 2)`](https://pkg.go.dev/strings#SplitN) as I think you intent to split into two parts, aren't you?

Reply:
I would also already assign `node` before first `if`. 

While we're here, maybe better also change `s := strings.SplitN(oid, "::", 2)`.

Reply:
Flipped the logic and changed both to `string.SplitN()`. I didn't know about splitN but it is super useful. Thank you for the suggestion!